cookbooks/bcpc-hadoop/recipes/disks.rb

```diff
@@ -1,57 +1,121 @@
-# vim: tabstop=2:shiftwidth=2:softtabstop=2 
-package "xfsprogs" do
-  action :install
+# vim: tabstop=2:shiftwidth=2:softtabstop=2
+#
+# Cookbook Name:: bcpc-hadoop
+# Recipe:: disks
+#
+# Copyright 2014-2016, Bloomberg Finance L.P.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+['xfsprogs', 'parted'].each do |package_name|
+  package package_name do
+    action :upgrade
+  end
 end
 
-reservation_requests = node[:bcpc][:hadoop][:disks][:reservation_requests]
-available_disks = node[:bcpc][:hadoop][:disks][:available_disks]
-role_min_disk = node[:bcpc][:hadoop][:disks][:role_min_disk]
+reservation_requests =
+  node[:bcpc][:hadoop][:disks][:reservation_requests]
+
+available_disks =
+  node[:bcpc][:hadoop][:disks][:available_disks]
 
-directory "/disk" do
-  owner "root"
-  group "root"
+role_min_disk =
+  node[:bcpc][:hadoop][:disks][:role_min_disk]
+
+directory '/disk' do
+  owner 'root'
+  group 'root'
   mode 00755
   action :create
 end
 
-if available_disks.length > 0 then
+if !available_disks.empty?
   available_disks.each_index do |i|
     directory "/disk/#{i}" do
-      owner "root"
-      group "root"
+      owner 'root'
+      group 'root'
       mode 00755
       action :create
       recursive true
     end
-   
-    d = available_disks[i]
-    execute "mkfs -t xfs -f /dev/#{d}" do
-      not_if "file -s /dev/#{d} | grep -q 'SGI XFS filesystem'"
+
+    base_name = available_disks[i]
+
+    #
+    # If the raw disk has already been formatted with an FS, no
+    # partition table is necessary.
+    #
+    # Otherwise, create a partition and format that.
+    #
+    require 'mixlib/shellout'
+
+    fs_check =
```

Comment:
This is an extension of code that was merged long ago.  The extended version has the same failure mode as the pre-existing code: the chef run fails.